token-swap/program/src/processor.rs

```diff
@@ -992,6 +1048,39 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes a [SetFreezeAuthorityBitMask](enum.Instruction.html).
+    pub fn process_set_freeze_authority_bit_mask(
+        _: &Pubkey,
+        freeze_authority_bit_mask: u8,
+        accounts: &[AccountInfo],
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let swap_info = next_account_info(account_info_iter)?;
+        let freeze_authority_info = next_account_info(account_info_iter)?;
+        let token_swap = SwapVersion::unpack(&swap_info.data.borrow())?;
+
+        Self::check_allowed_to_freeze(token_swap.as_ref(), freeze_authority_info)?;
+
+        let clone = SwapVersion::SwapV2(SwapV2 {
```

Comment:
This can become brittle once we add a v3 -- could you deserialize it and set the new bitmask through a function?  The v1 swap can error out for it